cmd/snap-confine/snap-confine.c

```diff
@@ -96,6 +97,157 @@ static void sc_maybe_fixup_udev(void)
 	}
 }
 
+/**
+ * sc_preservation_context remembers clobbered state to restore.
+ *
+ * The umask is preserved and restored to ensure consistent permissions for
+ * runtime system. The value is preserved and restored perfectly.
+ *
+ * The current working directory is preserved and possibly restored, if it
+ * exists in the runtime execution environment. The limitation on the current
+ * working directory is that not all paths that exist on the host filesystem
+ * exist in the execution environment. When the current working directory
+ * cannot be restored then a special substitute of /var/lib/snapd/void is used
+ * instead.
+**/
+typedef struct sc_preservation_context {
+	mode_t orig_umask;
+	int orig_cwd_fd;
+} sc_preservation_context;
+
+/**
+ * sc_remember_preservation context stores values to restore later.
+ *
+ * In addition the function makes the following changes to the process:
+ *  - the umask is set to 0
+ *  - the current working directory is set to /
+**/
+static void sc_remember_preservation_context(sc_preservation_context * ctx)
```

Comment:
Thinking about the name here, since it says `remember`, but there is nothing about the actual changes that are made. Maybe `sc_sanitize_current_context`?

Reply:
I think I agree with that remark.   sc_sanitize_process_state (see my other naming wondering in the previous PR)

Reply:
I'll follow up in the original discussion.